src/Kiota.Builder/Extensions/CodeComposedTypeBaseExtensions.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Linq;
+using Kiota.Builder.CodeDOM;
+using static Kiota.Builder.Writers.TypeScript.TypeScriptConventionService;
+
+namespace Kiota.Builder.Extensions;
+
+public static class CodeComposedTypeBaseExtensions
+{
+    public static bool IsComposedOfPrimitives(this CodeComposedTypeBase composedType)
```

Comment:
why define this as an extension method rather than in the CodeComposedTypeBase directly?

Reply:
No particular reason I can move it to the method